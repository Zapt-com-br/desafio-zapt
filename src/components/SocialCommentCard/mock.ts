export default {
  content:
    '”Não da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um só dia... O pessoal aqui de casa ficou viciado de um nível que terei que comprar toda semana... Assim não tem academia que secura rsrsrsrs Obrigado Zapt.”',
  user: {
    profileIcon: '/img/users-icon/user-icon1.png',
    username: '@ferri-gui'
  }
}
