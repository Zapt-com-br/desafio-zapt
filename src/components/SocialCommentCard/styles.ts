import styled, { css } from 'styled-components'

export const Card = styled.div`
  ${({ theme }) => css`
    width: 38.4rem;
    padding: ${theme.spacings.medium} ${theme.spacings.small};
    background-color: ${theme.colors.lightBlueBg};
    border-radius: ${theme.border.radius};
  `}
`

export const UserImage = styled.img``

export const UserProfile = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    margin-top: ${theme.spacings.small};
    gap: ${theme.spacings.xsmall};
  `}
`
