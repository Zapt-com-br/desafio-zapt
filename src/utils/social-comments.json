{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
