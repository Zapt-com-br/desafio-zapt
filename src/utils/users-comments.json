{
  "title": "O que as pessoas estão achando",
  "comments": [
    {
      "id": 1,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 2,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    },
    {
      "id": 3,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 4,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    }
  ]
}
