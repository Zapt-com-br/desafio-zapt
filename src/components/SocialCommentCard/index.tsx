import * as S from './styles';

export type SocialCommentCardProps = {
  comment: string;
  authorAvatarUrl: string;
  authorNickname: string;
};

const SocialCommentCard = ({
  comment,
  authorAvatarUrl,
  authorNickname,
}: SocialCommentCardProps) => (
  <S.Wrapper>
    <S.Comment>{comment}</S.Comment>

    <S.AuthorInfoContent>
      <S.AuthorAvatarWrapper>
        <S.AuthorAvatar
          src={authorAvatarUrl}
          alt={authorNickname}
          layout="fill"
          placeholder="blur"
          blurDataURL={authorAvatarUrl}
        />
      </S.AuthorAvatarWrapper>

      <S.AuthorNickname>{authorNickname}</S.AuthorNickname>
    </S.AuthorInfoContent>
  </S.Wrapper>
);

export default SocialCommentCard;
