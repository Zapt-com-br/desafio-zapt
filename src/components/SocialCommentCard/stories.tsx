import { Story, Meta } from '@storybook/react'

import comment from './mock'
import SocialCommentCard, { SocialCommentCardProps } from '.'

export default {
  title: 'SocialCommentCard',
  component: SocialCommentCard,
  args: {
    comment
  }
} as Meta

export const Default: Story<SocialCommentCardProps> = (args) => (
  <SocialCommentCard {...args} />
)
