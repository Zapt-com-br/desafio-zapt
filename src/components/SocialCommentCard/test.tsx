import { render, screen } from 'utils/test-utils'

import comment from './mock'
import SocialCommentCard from '.'

describe('<SocialCommentCard />', () => {
  it('should render with right username', () => {
    render(<SocialCommentCard comment={comment} />)

    expect(screen.getByText('@ferri-gui')).toBeInTheDocument()
  })

  it('should render user comment', () => {
    render(<SocialCommentCard comment={comment} />)

    expect(
      screen.getByText(
        '”Não da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um só dia... O pessoal aqui de casa ficou viciado de um nível que terei que comprar toda semana... Assim não tem academia que secura rsrsrsrs Obrigado Zapt.”'
      )
    ).toBeInTheDocument()
  })

  it('should render user picture', () => {
    render(<SocialCommentCard comment={comment} />)

    expect(screen.getByRole('img')).toHaveAttribute(
      'src',
      '/img/users-icon/user-icon1.png'
    )
  })
})
