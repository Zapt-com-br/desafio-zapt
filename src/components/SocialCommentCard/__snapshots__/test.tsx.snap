// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialCommentCard /> should render with comment 1`] = `
.c0 {
  background-color: #e9f5fe;
  border-radius: 0.4rem;
  padding: 2.4rem 1.6rem;
}

.c1 {
  color: #011fbb;
  font-weight: 400;
}

.c2 {
  margin-top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  position: relative;
  width: 3.6rem;
  height: 3.6rem;
  border-radius: 50%;
  background-color: #011fbb;
  border: 0.15rem solid #011fbb;
}

.c4 {
  border-radius: 50%;
}

.c5 {
  margin-left: calc(1.6rem - 0.4rem);
}

<div
  class="c0"
>
  <q
    class="c1"
  >
    Não da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um só dia... O pessoal aqui de casa ficou viciado de um nível que terei que comprar toda semana... Assim não tem academia que secura rsrsrsrs Obrigado Zapt.
  </q>
  <footer
    class="c2"
  >
    <div
      class="c3"
    >
      <img
        alt="@ferri-gui"
        class="c4"
        layout="fill"
        src="https://i.imgur.com/PZbgtl8.png"
      />
    </div>
    <p
      class="c5"
    >
      @ferri-gui
    </p>
  </footer>
</div>
`;
